pakyow-core/lib/pakyow/core/controller.rb

```diff
@@ -53,8 +55,8 @@ def process
       request.error = error
 
       catch :halt do
-        app.state_for(:router).each do |router|
-          if status = router.exception(error.class, context: self, handlers: handlers, exceptions: exceptions)
+        if failed_router
```

Comment:
If we do it this way, though, we will need to set `@current_router` to `nil` after the router is `call`ed. Otherwise @current_router will be always be set, even if the error comes from pakyow code.